tests/lib/rules/prop-types.js

```diff
@@ -2366,6 +2386,38 @@ ruleTester.run('prop-types', rule, {
       }]
     }, {
       code: [
+        'var Hello = function(props) {',
+        '  return <div>{props.firstname}</div>;',
+        '};',
+        'Hello.propTypes = {}'
+      ].join('\n'),
+      options: [{skipUndeclared: true}],
+      parserOptions: parserOptions,
+      errors: [{
+        message: '\'firstname\' is missing in props validation',
+        line: 2,
+        column: 22
+      }]
+    }, {
+      code: [
+        'class Hello extends React.Component {',
+        '  static get propTypes() {',
```

Comment:
Yes, I'm not sure why we have the getter version, and the "fourth" type is in fact the most standard. The third type there isn't actually valid because of the `static` keyword, but I think you mean its use in `React.createClass`, which is fine.